src/hardware/soft_limiter.cpp

```diff
@@ -0,0 +1,300 @@
+/*
```

Comment:
This file does not belong to `hardware` module. No hardware is being emulated here. Please move it to `misc` module.

Reply:
Done! (last commit in round 2)